### PR TITLE
Implement Error::cause() for BodyError

### PR DESCRIPTION
**Pull request by dotdash**

### CODE REVIEW

File: src/errors.rs [hunk 1]
```diff
@@ -22,6 +22,16 @@ impl StdError for BodyError {
     fn description(&self) -> &str {
         &self.detail[..]
     }
+
+    fn cause(&self) -> Option<&StdError> {
+        use BodyErrorCause::*;
+
+        match self.cause {
```
Comment (untitaker) [inline src/errors.rs]: I think you can do `self.cause.0` instead of the match clause?

Comment (dotdash) [inline src/errors.rs]: No, `.0` syntax only works for structs and tuple structs, but `cause` is an enum. What would the type of `self.cause.0` be? It depends on which variant you have.

**Comment by untitaker** [comment]
Thanks!

File: src/errors.rs (hunk 1 above)
Comment (untitaker) [inline src/errors.rs]: I would have expected it to infer that from the typesig, and auto-check that all variants impl that trait.

**Pull request merged**